frontend/src/identity/index.js

```diff
@@ -0,0 +1,107 @@
+import CeramicClient from "@ceramicnetwork/http-client";
+import ThreeIdResolver from "@ceramicnetwork/3id-did-resolver";
+import {IDX} from "@ceramicstudio/idx";
+import {EthereumAuthProvider, ThreeIdConnect} from "@3id/connect";
+import {DID} from "dids";
+
+import {definitions} from "./config.json";
+
+const ceramicProvider = CeramicClient.default ? CeramicClient.default : CeramicClient;
+const threeIdProvider = ThreeIdResolver.default
+  ? ThreeIdResolver.default
+  : ThreeIdResolver;
+
+async function authenticatedClient({
+  endpoint = "https://ceramic-clay.3boxlabs.com",
+  resolvers = null,
+  address = "",
+  provider = null,
+  ceramicClient = null,
+} = {}) {
+  let ceramic;
+  let ethereum = window.ethereum;
+
+  if (!ethereum)
+    return {
+      error: "No ethereum wallet detected",
+    };
+
+  if (!ceramicClient) {
+    ceramic = new ceramicProvider(endpoint);
+  } else {
+    ceramic = ceramicClient;
+  }
+
+  if (!resolvers) {
+    resolvers = {
+      ...threeIdProvider.getResolver(ceramic),
+    };
+  } else {
+    resolvers = resolvers.reduce((acc, next) => {
```

Comment:
I don't understand this neither, what does it do ?

What is the difference with the previous way of creating the resolver: 

```javascript
  const keyDidResolver = KeyDidResolver.getResolver();
  const threeIdResolver = ThreeIdResolver.getResolver(ceramic);
  const resolverRegistry = {
    ...threeIdResolver,
    ...keyDidResolver,
  };
  ```

Reply:
Regarding the reduce function.
As I understand it, when instantiating the `resolver` without passing the ceramic instance, we might've run into an issue where they're not bound to each other, thus the `.call(this, ceramic)`.

It's basically prepared to having a single instance of the resolver but, I didn't get that far to make sure it's efficient and not creating several instances.
I think we can make it more efficient in the future, but I don't yet 100% understand ceramic to be able to make such recommendations. But it works at the moment and I wouldn't disagree to simplifying it back to what it was, to make it clearer even if (maybe) it creates more instances of the `resolver` than we need.